packages/loader/container-definitions/src/loader.ts

```diff
@@ -285,13 +285,55 @@ export enum LoaderHeader {
     version = "version",
 }
 
+export interface IContainerLoadMode {
+    opsBeforeReturn?:
+        /*
+         * No trailing ops are applied before container is returned.
+         * Default value.
+         */
+        | undefined
+        /*
+         * Only cached trailing ops are applied before returning container.
+         * Caching is optional and could be implemented by the driver.
+         * If driver does not implement any kind of local caching strategy, this is same as above.
+         * Driver may cache a lot of ops, so care needs to be exercised (see below).
+         */
+        | "cached"
```

Comment:
how do datastores behave in this case. the ops will just be put into in memory cache at the data store layer, and not be applied util it is loaded

Reply:
Are you asking about "cached"? Correct. From consumer POV though, they are applied, as container moved to latest seq number and any loaded object would be at that seq number. In most cases developers would not be able to tell the difference, as we apply them as part of loading data store before user code (data object in particular) is running. So it's also faster that way (no event handlers installed yet)